src/components/inputs/date-input/date-input.js

```diff
@@ -25,267 +25,237 @@ import {
+    setCalendarDate(today);
+    setHighlightedIndex(suggestedItems.length + getDateInMonth(today) - 1);
+    inputRef.current.focus();
   };
 
-  emit = value =>
-    this.props.onChange({
-      target: {
-        id: this.props.id,
-        name: this.props.name,
-        // when cleared the value is null, but we always want it to be an
-        // empty string when there is no value.
-        value: value || '',
-      },
-    });
+  const jumpMonth = amount => {
+    const nextDate = changeMonth(calendarDate, amount);
+    setCalendarDate(nextDate);
+    setHighlightedIndex(0);
+  };
 
-  render() {
-    return (
-      <Constraints.Horizontal constraint={this.props.horizontalConstraint}>
-        <Downshift
-          key={this.props.intl.locale}
-          inputId={this.props.id}
-          itemToString={createItemToString(this.props.intl.locale)}
-          selectedItem={this.props.value === '' ? null : this.props.value}
-          highlightedIndex={this.state.highlightedIndex}
-          onChange={this.handleChange}
-          onStateChange={changes => {
-            /* eslint-disable no-prototype-builtins */
-            if (changes.hasOwnProperty('inputValue')) {
-              // input changed because user typed
-              if (changes.type === Downshift.stateChangeTypes.changeInput) {
-                const date = parseInputToDate(
-                  changes.inputValue,
-                  this.props.intl.locale
-                );
-                this.setState(
-                  date === ''
-                    ? {
-                        suggestedItems: [],
-                        highlightedIndex: null,
-                      }
-                    : {
-                        suggestedItems: [date],
-                        highlightedIndex: getDateInMonth(date) - 1,
-                        calendarDate: date,
-                      }
-                );
+  return (
+    <Constraints.Horizontal constraint={props.horizontalConstraint}>
+      <Downshift
+        key={intl.locale}
+        inputId={props.id}
+        itemToString={createItemToString(intl.locale)}
+        selectedItem={props.value === '' ? null : props.value}
+        highlightedIndex={highlightedIndex}
+        onChange={handleChange}
+        onStateChange={changes => {
+          /* eslint-disable no-prototype-builtins */
+          if (changes.hasOwnProperty('inputValue')) {
+            // input changed because user typed
+            if (changes.type === Downshift.stateChangeTypes.changeInput) {
+              const date = parseInputToDate(changes.inputValue, intl.locale);
+              if (date === '') {
+                setSuggestedItems([]);
+                setHighlightedIndex(null);
               } else {
-                // input changed because user selected a date
-                this.setState({
-                  suggestedItems: [],
-                  highlightedIndex: null,
-                });
+                setSuggestedItems([date]);
```

Comment:
I'm wondering if it makes sense to "batch" these updates? I guess one choice is to go with a "useReducer" but I feel like the readability is a bit lost then.

Reply:
Won‘t React batch them? Not 100% sure but maybe.